src/components/share/modal/ShareModal.tsx

```diff
@@ -0,0 +1,378 @@
+      groupName: t("components.share.search.group_name"),
+      elements: searchMemberResult ?? [],
+      showWhenEmpty: true,
+      emptyString,
+      endActions: isInvitationMode
+        ? [
+            {
+              content: <SearchUserItem user={newUser} />,
+              onSelect: () => void onSelect(newUser as UserData<UserType>),
+            },
+          ]
+        : undefined,
+    };
+    return group;
+  }, [searchUsersResult, searchQuery, t, pendingInvitationUsers, onSelect]);
+
+  /**
+   * Set the height of the list of the quick search content.
+   * Because the package sets the size automatically based on the content
+   */
+  const handleRef = (node: HTMLDivElement) => {
+    const inputHeight = 70;
+    const footerHeight = node?.clientHeight ?? 0;
+    const selectedUsersHeight = selectedUsersRef.current?.clientHeight ?? 0;
+    const height = `calc(${modalContentHeight} - ${footerHeight}px - ${selectedUsersHeight}px - ${inputHeight}px - 10px)`;
+    setListHeight(height);
+  };
+
+  return (
+    <Modal
+      title={props.modalTitle ?? t("components.share.modalTitle")}
+      isOpen={props.isOpen}
+      onClose={props.onClose}
+      closeOnClickOutside
+      size={isMobile ? ModalSize.FULL : ModalSize.LARGE}
+    >
+      <div>
+        {pendingInvitationUsers.length > 0 && (
+          <div
+            className="c__share-modal__selected-users"
+            ref={selectedUsersRef}
+          >
+            <InvitationUserSelectorList
+              users={pendingInvitationUsers}
+              onRemoveUser={onRemoveUser}
+              roles={props.invitationRoles}
+              selectedRole={selectedInvitationRole}
+              onSelectRole={setSelectedInvitationRole}
+              onShare={() => {
+                props.onInviteUser(
+                  pendingInvitationUsers,
+                  selectedInvitationRole
+                );
+                setPendingInvitationUsers([]);
+              }}
+            />
+          </div>
+        )}
+        <QuickSearch
+          onFilter={onInputChange}
+          inputValue={inputValue}
+          loading={props.loading}
+          placeholder={t("components.share.user.placeholder")}
+        >
+          <div
+            style={{
+              height: listHeight,
+              overflowY: "auto",
+            }}
+          >
+            {showSearchUsers && !props.loading && (
+              <div className="c__share-modal__search-users">
+                <QuickSearchGroup
+                  group={usersData}
+                  onSelect={(user) => {
+                    onSelect(user);
+                  }}
+                  renderElement={(user) => <SearchUserItem user={user} />}
+                />
+              </div>
+            )}
+
+            {!showSearchUsers && !props.loading && children}
+
+            {/* Invitations list */}
+            {!showSearchUsers && !props.loading && invitations.length > 0 && (
```

Comment:
would be nice to have a named variable like `showInvitations` that contains the result of this condition, it improve code readability